ansible/roles/test/tasks/warm-reboot-fib.yml

```diff
@@ -54,8 +54,7 @@
   when: warm_restart_docker is defined
 
 - name: reboot dut
-  shell: reboot
-  become: true
+  include: common_tasks/reboot_sonic.yml
```

Comment:
calling reboot will cancel the previous scheduled warm reboot. if the intention here is to trigger a warm reboot, then you need to set the reboot type before including reboot_sonic.yml

Reply:
but I didn't change the reboot type. In fact, I just add wait for switch to be up after the reboot.
In my tests this command simple run a switch into reboot and that's it.
But once `warm-reboot-fib.yml` is finishing  Ansible starts post-testcase checks and is trying to check various facts on the DUT (e.g. syncd state) and of course it fails.

Reply:
@praveen-li can you clarify the intention of this reboot here? Do we need to cause a warm reboot or code reboot?

Reply:
@yxieca 
I wrote this test while warm-reboot war room. As per this, we set "config warm_restart enable system" as below and reboot the system for warm-reboot. If any other command is used for warm-reboot that must be used here. Basically the assumption was reboot perform warm-reboot after setting correct config.
---------
- name: set config warm_restart enable system
  shell: config warm_restart enable system
  become: true
  when: warm_restart_docker is not defined

Reply:
Thanks @praveen-li for clarification. 'reboot' is intended to do normal cold reboot. In fact it will disable warm reboot flag before proceeding. And 'warm-reboot' script will automatically set the warm reboot flag before proceeding too.

So this test could be simplified by removing the warm-reboot flag setting.

If it is container based warm_restart, some container, e.g. swss, requires swss warm reboot flag to be set. So there are extra works here too.

If we are not doing container based warm reboot, then set the reboot_type to warm-reboot before calling sonic_reboot is sufficient.

Reply:
have updated the script

Reply:
have checked using the last codechange - the test passed